www/js/goals.js

```diff
@@ -521,40 +520,78 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
             $scope.topThreeUsers.push(threeUsers[0]);
             $scope.topThreeUsers.push(threeUsers[2]);
             $scope.usersUnderTopThree = users.slice(3);
-            //allPartyList = getPatyForAllUsers(users);
             console.log($scope.topThreeUsers);
-            //console.log(allPartyList);
+            getPartyForAllUsers(users);
+            setRank(partyList);
         }, function(error){
             console.log("Error getting all the users");
             console.log(error);
         });
     };
 
-    /*var getPatyForAllUsers = function(users) {
-        var partys = [];
+    var getPartyForAllUsers = function(users) {
+        partyList = [];
         for (var i = 0; i < users.length; i++) {
+            var partyInList = false;
             if('_id' in users[i].party){
-                userParty = getPartyById(users[i].party._id);
-                partys.push(userParty);
-            } else {
-                var noPartyObj = {'name': 'Has not joined a Party'};
-                partys.push(noPartyObj);
+                if(partyList.length === 0){
+                    var party = {'_id' : users[i].party._id, 'members': [users[i]], 'partyObj': {}, 'stats':{}};
+                    getPartyById(users[i].party._id, 0);
+                    partyList.push(party);
+                } else {
+                    for (var j = 0; j < partyList.length; j++) {
+                        if(partyList[j]._id === users[i].party._id){
+                            partyList[j].members.push(users[i]);
+                            partyInList = true;
+                        }
+                    }
+                    if(!partyInList) {
+                        party = {'_id' : users[i].party._id, 'members': [users[i]], 'partyObj': {}, 'stats':{}};
+                        getPartyById(users[i].party._id, j);
+                        partyList.push(party);
+                    }
+                }
             }
         }
-        return partys;
     };
 
-    var getPartyById = function(id) {
-        var callOpts = {'method': 'GET', 'method_url': "/api/v3/groups/" + id,
+    var getPartyById = function(id, j) {
+        var callOpts = {'method': 'GET', 'method_url': "/api/v3/groups/"+id,
                     'method_args': null};
         CommHelper.habiticaProxy(callOpts).then(function(response){
-            console.log("Sucessfully a user's party");
-            var partyObj = response.data;
-            return partyObj;
+            console.log("Sucessfully got a user's party");
+            partyList[j].partyObj = response.data;
```

Comment:
nit: to make this more general, you might want to pass the `partyList[j]` object instead passing in `j` and using it as an index to the list. Reduces the dependency between this function and the representation of the `partyList`